osu.Game/Rulesets/Difficulty/DifficultyCalculator.cs

```diff
@@ -112,11 +115,12 @@ public IEnumerable<DifficultyAttributes> CalculateAll()
         /// Performs required tasks before every calculation.
         /// </summary>
         /// <param name="mods">The original list of <see cref="Mod"/>s.</param>
-        private void preProcess(Mod[] mods)
+        /// <param name="cancellationToken">The cancellation cancellationToken.</param>
+        private void preProcess(IEnumerable<Mod> mods, CancellationToken cancellationToken = default)
         {
             playableMods = mods.Select(m => m.DeepClone()).ToArray();
```

Comment:
Should handle null `mods` and set `playableMods` to an empty array instead.

Reply:
@frenzibyte would prefer that that was handled once at the top-level and not in each leaf method as i am guessing you want it to be from where you placed this comment.

Reply:
Each way works I think, don't have any strong opinion on that.

Reply:
I'm confused. Why is this being changed in this PR?

Reply:
Seems that got answered in https://github.com/ppy/osu/pull/15494#discussion_r744417289.